src/main/java/net/dv8tion/jda/internal/handle/ChannelDeleteHandler.java

```diff
@@ -90,19 +90,25 @@ protected Long handleInternally(DataObject content)
                     return null;
                 }
 
-                // This is done in the AudioWebSocket already
-//                //We use this instead of getAudioManager(Guild) so we don't create a new instance. Efficiency!
-//                AudioManagerImpl manager = (AudioManagerImpl) getJDA().getAudioManagersView().get(guild.getIdLong());
-//                if (manager != null && manager.isConnected()
-//                        && manager.getConnectedChannel().getIdLong() == channel.getIdLong())
-//                {
-//                    manager.closeAudioConnection(ConnectionStatus.DISCONNECTED_CHANNEL_DELETED);
-//                }
+                //  This is done in the AudioWebSocket already
+                //  We use this instead of getAudioManager(Guild) so we don't create a new instance. Efficiency!
+                //  AudioManagerImpl manager = (AudioManagerImpl) getJDA().getAudioManagersView().get(guild.getIdLong());
+                //  if (manager != null && manager.isConnected()
+                //          && manager.getConnectedChannel().getIdLong() == channel.getIdLong())
+                //  {
+                //      manager.closeAudioConnection(ConnectionStatus.DISCONNECTED_CHANNEL_DELETED);
+                //  }
                 guild.getVoiceChannelsView().remove(channel.getIdLong());
                 getJDA().handleEvent(
-                    new ChannelDeleteEvent(
-                        getJDA(), responseNumber,
-                        channel));
+                        new ChannelDeleteEvent(
+                                getJDA(), responseNumber,
+                                channel));
+
+                // Deleting any guild scheduled events associated to the deleted channel as they are deleted when the channel gets deleted.
+                // There is no delete event for the deletion of guild scheduled events, so we do this to keep the cache in sync.
+                guild.getScheduledEventsView().stream()
```

Comment:
Does discord delete them too? Or is it possible to still change the location after the fact

Reply:
yes, if an event that takes place inside a Voice or Stage Channel and the channel gets deleted, Discord also deletes the event.